pilot/pkg/serviceregistry/kube/controller/endpoint_builder.go

```diff
@@ -76,6 +76,8 @@ func NewEndpointBuilder(c controllerInterface, pod *v1.Pod) *EndpointBuilder {
 		namespace:    namespace,
 		hostname:     hostname,
 		subDomain:    subdomain,
+		// todo(kebe7jun) use ISTIO_META_NETWORK instead?
+		metaNetwork: network.ID(podLabels[label.TopologyNetwork.Name]),
```

Comment:
This is not correct, the original intention isto prefer the network from metadata which is from `ISTIO_META_NETWORK`. 

But it is changing to respect pod label first

Reply:
The implementation was changed.

Reply:
On the pods with this bug, what does there ISTIO_META value hold?

If it's different from the label that will cause issues.

If it's empty, we should only fallback to the label.